src/adr/0011-remove-elasticsearch-document-source.md

```diff
@@ -0,0 +1,33 @@
+# 11. Disable ElasticSearch dynamic mapping
+
+Date: 2019-10-17
+
+## Status
+
+Rejected
+
+The benefits do not outweigh the costs.
+
+## Context
+
+Though very handy to have around, the source field does incur storage overhead within the index. 
+
+## Decision
+
+Disable _source for ELasticSearch indexed documents.
+
+## Consequences
+
+We notice a 20% space reduction of data stored on ElasticSearch.
```

Comment:
Can you precise the context? This number seems strange, because it means that we are storing 5x more data on ElasticSearch than the raw data. It's huge!
I guess your emails are full of headers and text only, which is perhaps not representative of a modern usage? What about emails with attachments and HTML?

Reply:
> because it means that we are storing 5x more data on ElasticSearch than the raw data

We discussed that, we store (in space) 5 time the raw document, which is not that surprising given than some fields are indexed twice (for search & for sort), that we rely on nested objects.

There was a confusion between the original MIME document and the indexed document.

Reply:
`Given a dataset composed of small text/plain messages, we notice a 20% space reduction of data stored on ElasticSearch.`

Details added ;-)